tests/lax_scipy_test.py

```diff
@@ -157,6 +158,29 @@ def lax_fun(a):
                             tol={onp.float32: 1e-3, onp.float64: 1e-14})
     self._CompileAndCheck(lax_fun, args_maker)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_inshape={}_x={}".format(
+          jtu.format_shape_dtype_string(shape, dtype), x),
+       "rng_factory": jtu.rand_positive, "shape": shape, "dtype": dtype,
+       "x": x}
+      for shape in all_shapes
+      for dtype in float_dtypes
+      for x in [0, 1, 2, onp.array([1, 2, 3, 4])]))
+  def testPolygamma(self, rng_factory, shape, dtype, x):
+    def scipy_fun(q):
+      return osp_special.polygamma(x, q)
+
+    def lax_fun(q):
+      return lsp_special.polygamma(x, q)
+
+    if onp.ndim(x) == 1 and (len(shape) == 0 or shape[-1] != x.shape[-1]):
+      raise SkipTest("arguments are not broadcastable")
+    rng = rng_factory(self.rng())
+    args_maker = lambda: [rng(shape, dtype)]
+    self._CheckAgainstNumpy(scipy_fun, lax_fun, args_maker,
+                            tol={onp.float32: 1e-6, onp.float64: 1e-12})
+    self._CompileAndCheck(lax_fun, args_maker)
```

Comment:
Can you test `polygamma` as part of `JAX_SPECIAL_FUNCTION_RECORDS`? I believe it only tests first-order derivatives.

Reply:
Currently, `JAX_SPECIAL_FUNCTION_RECORDS` does not generate both integer and float args. I'll try to see if adding several if/else statements to `testScipySpecialFun` is simpler.

Reply:
Thanks for suggesting, @jakevdp ! By using `JAX_SPECIAL_FUNCTION_RECORDS` to test first-order derivatives, I observed an issue with the old jvp rule `lambda g, ans, x, y: g * f(x, y)`, which is invalid when `x` and `y` have different shapes (in that case, f(x, y) will have broadcasted shape, while `g` keeps the same shape as `y`. So I have split `polygamma` into a custom jvp `_polygamma` whose arguments have the same shape.